### PR TITLE
fix Output and Input comparison to use instance id instead of addresses

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 - fix Output and Input comparison operators
 - addresses differ on each run instead of instance id

### Tickets:
 - 118730

### CODE REVIEW

File: src/core/src/node_input.cpp [hunk 1]
```diff
@@ -61,11 +61,13 @@ bool Input<Node>::operator!=(const Input& other) const {
     return !(*this == other);
 }
 bool Input<Node>::operator<(const Input& other) const {
-    return m_node < other.m_node || (m_node == other.m_node && m_index < other.m_index);
```
Comment (itikhono) [inline src/core/src/node_input.cpp:64]: also for `operator == `

Comment (evkotov) [reply src/core/src/node_input.cpp:64]: Is it needed? if addresses are equal, than instance id also should be equal? Or not?

Comment (evkotov) [reply src/core/src/node_input.cpp:64]: In operators< and > I also don't use instance id for checking equivalence

Comment (itikhono) [reply src/core/src/node_input.cpp:64]: ok, agree

**Comment by itikhono** [comment]
Looks good, CI is green
Could you add a new test?

**Comment by evkotov** [comment]
This experiment is seemed to be successfull.

**Pull request closed without merging**